### PR TITLE
Sorting

### LINKED ISSUE #509: `Modified On` sorting does not work for Applications

### LINKED ISSUE #538: state sorting do not work in tasks and applications
Screenshot 2021-12-06 at 14 24 09

### PR DESCRIPTION
**Pull request by BykhovDenis**
close #509 close #538 Signed-off-by: Denis Bykhov [bykhov.denis@gmail.com](mailto:bykhov.denis@gmail.com)

### CODE REVIEW

File: packages/core/src/query.ts [hunk 1]
```diff
@@ -75,7 +75,15 @@ function arrayOrValue (vv: any): any[] {
 export function resultSort<T extends Doc> (result: T[], sortOptions: SortingQuery<T>): void {
   const sortFunc = (a: any, b: any): number => {
     for (const key in sortOptions) {
-      const result = typeof a[key] === 'string' ? a[key].localeCompare(b[key]) : a[key] - b[key]
+      let aValue = getNestedValue(key, a)
+      if (typeof aValue === 'object') {
+        aValue = JSON.stringify(aValue)
```
Comment (haiodo) [inline packages/core/src/query.ts:80]: Why we need to convert to JSON?

Comment (BykhovDenis) [reply packages/core/src/query.ts:80]: Simple way for compare objects. Let me know if you have any ideas

Comment (haiodo) [reply packages/core/src/query.ts:80]: Lets's define sorters based on type, or use some contribution logic. It is not good if object for example has number, it not work well.

Comment (BykhovDenis) [reply packages/core/src/query.ts:80]: How can I use type-based soreters with MongoDB...?

Comment (BykhovDenis) [reply packages/core/src/query.ts:80]: I think we should not allow sorting by objects.

Comment (haiodo) [reply packages/core/src/query.ts:80]: Why we need this part of code in this case?

Comment (BykhovDenis) [reply packages/core/src/query.ts:80]: We can extend the `BuildModeKey` from #576 with the `sortKey` property, for example ...

**Review by aplatoff — APPROVED** [review]

**Pull request merged**